saleor/payment/utils.py

```diff
@@ -167,75 +234,89 @@ def gateway_authorize(payment: Payment, payment_token: str) -> Transaction:
     """
     clean_authorize(payment)
 
-    gateway, gateway_params = get_payment_gateway(payment.gateway)
-    with transaction.atomic():
-        txn, error = gateway.authorize(
-            payment=payment, payment_token=payment_token, **gateway_params)
-    if not txn.is_success:
-        raise PaymentError(error)
-    return txn
+    return call_gateway('authorize',
+        transaction_kind=TransactionKind.AUTH, payment=payment,
+        payment_token=payment_token)
 
 
 @validate_payment
-def gateway_capture(payment: Payment, amount: Decimal) -> Transaction:
+def gateway_capture(payment: Payment, amount: Decimal=None) -> Transaction:
     """Captures the money that was reserved during the authorization stage."""
+    if amount is None:
+        amount = payment.get_charge_amount()
     clean_capture(payment, amount)
 
-    gateway, gateway_params = get_payment_gateway(payment.gateway)
-    with transaction.atomic():
-        txn, error = gateway.capture(
-            payment=payment, amount=amount, **gateway_params)
-        if txn.is_success:
-            payment.charge_status = ChargeStatus.CHARGED
-            payment.captured_amount += txn.amount
-            payment.save(update_fields=['charge_status', 'captured_amount'])
-            order = payment.order
-            if order and order.is_fully_paid():
-                handle_fully_paid_order(order)
-    if not txn.is_success:
-        raise PaymentError(error)
-    return txn
+    auth_transaction = payment.transactions.filter(
```

Comment:
The ```auth_transaction``` could be None and it should be validated.

Reply:
My understanding was that this was already provided by the functions called above this. Like 'clean_capture' calls 'payment.can_capture' which in turn has a check for 'is_authorized'.

Reply:
If so, these helper needs to be updated. As far as I know, they didn't have checks on the transactions. They only check the payment status or charge amount.

Plus, there is no harm to recheck this to avoid exceptions.